iot/iot.rb

```diff
@@ -0,0 +1,795 @@
+  # project_id   = "Your Google Cloud project ID"
+  # location_id  = "The Cloud region that you created the registry in"
+  # registry_id  = "The Google Cloud IoT Core device registry identifier"
+  # pubsub_topic = "The Google Cloud PubSub topic to use for this registry"
+
+  require "google/apis/cloudiot_v1"
+
+  # Initialize the client and authenticate with the specified scope
+  Cloudiot = Google::Apis::CloudiotV1
+  iot_client = Cloudiot::CloudIotService.new
+  iot_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The project / location where the registry is created.
+  parent = "projects/#{project_id}/locations/#{location_id}"
+
+  registry = Cloudiot::DeviceRegistry.new
+  registry.id = registry_id
+  registry.event_notification_configs = [Cloudiot::EventNotificationConfig::new]
+  registry.event_notification_configs[0].pubsub_topic_name = pubsub_topic
+
+  registry = iot_client.create_project_location_registry(
+      parent, registry
+  )
+
+  puts "Created registry: #{registry.name}"
+  # [END iot_create_registry]
+end
+
+
+$delete_registry = -> (project_id:, location_id:, registry_id:) do
+  # [START iot_delete_registry]
+  # project_id  = "Your Google Cloud project ID"
+  # location_id = "The Cloud region that you created the registry in"
+  # registry_id = "The Google Cloud IoT Core device registry identifier"
+
+  require "google/apis/cloudiot_v1"
+
+  # Initialize the client and authenticate with the specified scope
+  Cloudiot = Google::Apis::CloudiotV1
+  iot_client = Cloudiot::CloudIotService.new
+  iot_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the project
+  resource = "projects/#{project_id}/locations/#{location_id}/registries/#{registry_id}"
+
+  result = iot_client.delete_project_location_registry(resource)
+  puts "Deleted registry: #{registry_id}"
+  # [END iot_delete_registry]
+end
+
+
+$get_iam_policy = -> (project_id:, location_id:, registry_id:) do
+  # [START iot_get_iam_policy]
+  # project_id  = "Your Google Cloud project ID"
+  # location_id = "The Cloud region the registry is located in"
+  # registry_id = "The registry to get an IAM policy for"
+
+  require "google/apis/cloudiot_v1"
+
+  # Initialize the client and authenticate with the specified scope
+  Cloudiot = Google::Apis::CloudiotV1
+  iot_client = Cloudiot::CloudIotService.new
+  iot_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the project
+  parent = "projects/#{project_id}/locations/#{location_id}"
+  resource = "#{parent}/registries/#{registry_id}"
+
+  # List the devices in the provided region
+  policy = iot_client.get_registry_iam_policy(
+    resource
+  )
+
+  puts policy
+  puts policy.inspect
```

Comment:
Is this left-over from debugging?

I'd prefer not to `puts object` or `puts object.inspect` because it's not useful prose, the reader doesn't know what this will output

Reply:
Yes, also found another where I was printing debugging.